protocol-tokio/src/protocol/codec/message.rs

```diff
@@ -88,17 +89,113 @@ pub enum Message {
     ProbeSocketAck(nt::LightWeightConnectionId),
     CreateNodeId(nt::LightWeightConnectionId, NodeId),
     AckNodeId(nt::LightWeightConnectionId, NodeId),
+    Bytes(nt::LightWeightConnectionId, Bytes),
+
+    UserMessage(M),
```

Comment:
This extension variant complicates the code quite a lot. I wonder if it's really needed.